### PR TITLE
Updated security manager SDK to support multiple policy mapping

### PR DESCRIPTION
**Pull request by larkinscarvalho**

### CODE REVIEW

File: src/main/java/org/osc/sdk/manager/Constants.java [hunk 1]
```diff
@@ -68,4 +68,9 @@
      * The name of the OSGi property that indicates whether the security manager is capable of synchronizing policy mapping with OSC.
      */
     public static final String SYNC_POLICY_MAPPING = "osc.plugin.manager.sync_policy_mapping";
+
+    /**
+     * The name of the OSGi property that indicates whether the security manager supports multiple tags.
```
Comment (emanoelxavier) [inline src/main/java/org/osc/sdk/manager/Constants.java]: tags -> policies

File: src/main/java/org/osc/sdk/manager/api/ManagerSecurityGroupInterfaceApi.java [hunk 2]
```diff
@@ -34,26 +35,21 @@
     /**
      * Creates a security group interface within context of the current virtual system
      *
-     * @param name  the name of the security group interface
-     * @param policyId  the identifier of the policy
-     * @param tag  the tag associated with the policy defined by OSC
-     * @return the identifier of the created policy
+     * @param sgiElem  provides context information of the SecurityGroupInterface
+     * @return the identifier of the security group interface
```
Comment (emanoelxavier) [inline src/main/java/org/osc/sdk/manager/api/ManagerSecurityGroupInterfaceApi.java]: ...interface defined by the security manager

File: src/main/java/org/osc/sdk/manager/api/ManagerSecurityGroupInterfaceApi.java [hunk 3]
```diff
@@ -34,26 +35,21 @@
     /**
      * Creates a security group interface within context of the current virtual system
      *
-     * @param name  the name of the security group interface
-     * @param policyId  the identifier of the policy
-     * @param tag  the tag associated with the policy defined by OSC
-     * @return the identifier of the created policy
+     * @param sgiElem  provides context information of the SecurityGroupInterface
```
Comment (emanoelxavier) [inline src/main/java/org/osc/sdk/manager/api/ManagerSecurityGroupInterfaceApi.java]: the information needed to create a security group interface.
Note:  Avoid using type names on the javadoc unless it is a link or a see. For other cases we should be cursive, i.e.: security group interface

File: src/main/java/org/osc/sdk/manager/api/ManagerSecurityGroupInterfaceApi.java [hunk 4]
```diff
@@ -34,26 +35,21 @@
...
+    String createSecurityGroupInterface(SecurityGroupInterfaceElement sgiElem) throws Exception;
 
     /**
      * Updates a security group interface within context of the current virtual system
      *
-     * @param id  the identifier of the security group interface to be updated
-     * @param name  the new name of the security group interface
-     * @param policyId  the identifier of the new policy
-     * @param tag  the new tag associated with the policy defined by OSC
+     * @param sgiElem  provides context information of the SecurityGroupInterface
```
Comment (emanoelxavier) [inline src/main/java/org/osc/sdk/manager/api/ManagerSecurityGroupInterfaceApi.java]: dito

File: src/main/java/org/osc/sdk/manager/element/ManagerSecurityGroupInterfaceElement.java [hunk 5]
```diff
@@ -34,10 +36,21 @@
      */
     String getName();
 
-    /**
-     * @return the identifier of this policy defined by the security manager
-     */
-    String getPolicyId();
+	/**
+	 * Provide manager security group id if the manager is supporting policy
```
Comment (emanoelxavier) [inline src/main/java/org/osc/sdk/manager/element/ManagerSecurityGroupInterfaceElement.java]: Provides the identifier of the security group defined by security managers that support policy mapping and security groups. 
Also please follow a similar text for other places

File: src/main/java/org/osc/sdk/manager/element/ManagerSecurityGroupInterfaceElement.java [hunk 6]
```diff
@@ -34,10 +36,21 @@
...
+	 * mapping and security group.
+	 *
+	 * @return the manager security group identifier
```
Comment (emanoelxavier) [inline src/main/java/org/osc/sdk/manager/element/ManagerSecurityGroupInterfaceElement.java]: the identifier of the security group defined by the manager

File: src/main/java/org/osc/sdk/manager/element/ManagerSecurityGroupInterfaceElement.java [hunk 7]
```diff
@@ -34,10 +36,21 @@
...
+	 */
+	String getManagerSecurityGroupId();
+
+	/**
+	 * Provide manager policy id in case of IP/MAC or encapsulation tag in case
```
Comment (emanoelxavier) [inline src/main/java/org/osc/sdk/manager/element/ManagerSecurityGroupInterfaceElement.java]: Providers the identifier of the policies supported by the manager. If the appliance uses encapsulation tags this should be the tags supported by the manager.

File: src/main/java/org/osc/sdk/manager/element/ManagerSecurityGroupInterfaceElement.java [hunk 8]
```diff
@@ -34,10 +36,21 @@
...
+	 * security appliance supports encapsulation.
+	 *
+	 * @return the set of policy id's
```
Comment (emanoelxavier) [inline src/main/java/org/osc/sdk/manager/element/ManagerSecurityGroupInterfaceElement.java]: id ->  identifiers (also check other places). Also maybe:
the set of policy identifiers or encapsulation tags supported by this manager

File: src/main/java/org/osc/sdk/manager/element/SecurityGroupInterfaceElement.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.sdk.manager.element;
+
+import java.util.Set;
+
+public interface SecurityGroupInterfaceElement {
+
+	/**
+	 * Id of the security group interface.
```
Comment (emanoelxavier) [inline src/main/java/org/osc/sdk/manager/element/SecurityGroupInterfaceElement.java]: for simple things like this u can remove the description part and simply just have the return part below. Having both with the same or very similar information is redundant.

File: src/main/java/org/osc/sdk/manager/element/SecurityGroupInterfaceElement.java [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+	 *
+	 * @return the identifier of the security group interface
+	 */
+	String getSecurityGroupInterfaceId();
+
+	/**
+	 * Name of the security group interface.
```
Comment (emanoelxavier) [inline src/main/java/org/osc/sdk/manager/element/SecurityGroupInterfaceElement.java]: dito

File: src/main/java/org/osc/sdk/manager/element/SecurityGroupInterfaceElement.java [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+	 */
+	String getManagerSecurityGroupId();
+
+	/**
+	 * Provide manager policy id in case of IP/MAC or encapsulation tag in case
+	 * security appliance supports encapsulation.
+	 *
+	 * @return the set of policy id's
+	 */
+	Set<String> getManagerPolicyIds();
```
Comment (emanoelxavier) [inline src/main/java/org/osc/sdk/manager/element/SecurityGroupInterfaceElement.java]: are u sure this is really the tags? If i remember correctly this is just the policy ids, for tags the `getTag` is used. Please update the javadoc throughout accordingly.

**Comment by larkinscarvalho** [comment]
@emanoelxavier I have addressed your comments. Also, updated spaces to tabs.

File: src/main/java/org/osc/sdk/manager/element/ManagerSecurityGroupInterfaceElement.java [hunk 12]
```diff
@@ -24,23 +26,33 @@
...
+	 * @return the identifier of the security group defined by the manager
+	 */
+	String getManagerSecurityGroupId();
 
-    /**
-     * @return the identifier of this policy defined by the security manager
-     */
-    String getPolicyId();
+	/**
+	 * Providers the identifier of the policies supported by the manager
```
Comment (emanoelxavier) [inline src/main/java/org/osc/sdk/manager/element/ManagerSecurityGroupInterfaceElement.java]: Provides

File: src/main/java/org/osc/sdk/manager/element/SecurityGroupInterfaceElement.java [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+
+	/**
+	 * Provides the identifier of the policies supported by the manager
+	 *
+	 * @return the set of policy identifiers
+	 */
+	Set<String> getManagerPolicyIds();
+
+	/**
+	 *
```
Comment (emanoelxavier) [inline src/main/java/org/osc/sdk/manager/element/SecurityGroupInterfaceElement.java]: extra ln

**Pull request merged**